source/geometries/Next100FieldCage.cc

```diff
@@ -613,6 +636,146 @@ void Next100FieldCage::BuildFieldCage()
+  G4double first_ring_drif_z_pos = drift_ring_dist_/2. + active_short_z/2. + gate_teflon_dist_ + GetELzCoord();
+  G4double buffer_short_z = 37.*mm;
+
+  G4double ring_drift_buffer_dist = 72.*mm;
+  G4int num_drift_rings = 48;
+  G4int num_buffer_rings = 4;
+  G4double posz;
+  G4double first_ring_buff_z_pos = first_ring_drif_z_pos + (num_drift_rings-1)*drift_ring_dist_ + ring_drift_buffer_dist;
+
+  G4Tubs* ring_solid =
+  new G4Tubs("RING", ring_int_diam_/2., ring_ext_diam_/2., ring_thickn_/2., 0, twopi);
+
+  G4LogicalVolume* ring_logic =
+  new G4LogicalVolume(ring_solid, copper_, "RING");
+
+
+
+  //Placement of the drift rings.
+  for (G4int i=0; i<num_drift_rings; i++) {
+    posz = first_ring_drif_z_pos + i*drift_ring_dist_;
+    new G4PVPlacement(0, G4ThreeVector(0., 0., posz), ring_logic,
+                      "RING", mother_logic_, false, i, true);
+  }
+
+  //Placement of the buffer rings.
+  for (G4int i=0; i<num_buffer_rings; i++) {
+    posz = first_ring_buff_z_pos + i*buffer_ring_dist_;
+    new G4PVPlacement(0, G4ThreeVector(0., 0., posz), ring_logic,
+                      "RING", mother_logic_, false, i, true);
+  }
+  // Ring holders.
+  // ACTIVE holders.
+  //Active_long are the base of the ring holders and active_short are the smaller pieces over the long one.
+  G4double active_long_x = 60*mm;
+  G4double active_long_y = 9*mm;
+  G4double active_short_x = active_long_x;
+  G4double active_short_y = 33.15*mm;
+  G4double r = (active_diam_+2*teflon_thickn_+active_long_y)/2.;
+  G4Box* active_short_solid =
+  new G4Box("ACT_SHORT", active_short_x/2., active_short_y/2., active_short_z/2.);
+
+  G4double first_act_short_z = -teflon_drift_length_/2.+ active_short_z/2.;
+
+  G4Box* active_long_solid =
+    new G4Box("ACT_LONG", active_long_x/2., active_long_y/2., teflon_drift_length_/2.);
+
+  G4UnionSolid* act_holder_solid =
+    new G4UnionSolid ("ACT_HOLDER", active_long_solid, active_short_solid, 0,
+                      G4ThreeVector(0.,active_long_y/2.+active_short_y/2.,first_act_short_z));
+
+  for (G4int j=1; j<num_drift_rings; j++) {
+    posz = first_act_short_z + j*drift_ring_dist_;
+
+    act_holder_solid = new G4UnionSolid("ACT_HOLDER", act_holder_solid, active_short_solid, 0,
+                                        G4ThreeVector(0.,active_long_y/2. + active_short_y/2.,posz));
+    }
+
+  G4LogicalVolume* act_holder_logic = new G4LogicalVolume(act_holder_solid,pe1000_,"ACT_HOLDER");
+  G4int numbering=0;
+  for (G4int i=10; i<360; i +=20){
+    //G4cout << "i " << i << G4endl;
+    //G4cout << "90-i " << 90-i << G4endl;
+    G4RotationMatrix* rot = new G4RotationMatrix();
+    rot -> rotateZ((90-i) *deg);
+    new G4PVPlacement(rot, G4ThreeVector(r*cos(i*deg), r*sin(i*deg), teflon_drift_zpos_), act_holder_logic,
+                                       "ACT_HOLDER", mother_logic_, false, numbering, true);
+    numbering +=1;}
+
+    // BUFFER holders.
+    G4double buffer_long_x = active_long_x;
```

Comment:
What's the point of redefining the variables if they are exactly the same? You could use a generic name instead, which illustrates both uses, like `holder_long_x`, for instance. This holds for `r`, as well.

Reply:
ok!